common/recommender/Baseline.js

```diff
@@ -284,7 +306,7 @@ class Baseline {
     let penalty = 0.8;
 
     return entries.map((entry, i, arr) => {
-      if (i > 0 && arr[i - 1].host === entry.host) {
+      if (i > 0 && this._similarItems(arr[i - 1], entry)) {
```

Comment:
I think this is cleaner and makes more sense:

```js
_similarItems(prev, curr) {
    const imgPrev = getBestImage(prev.images);
    const imgCurr = getBestImage(curr.images);
    const hasImage = imgPrev && imgCurr;
    return prev.host === curr.host || (hasImage && imgPrev.url === imgCurr.url);
  }
  
dedupe(entries) {
    let penalty = 0.8;
    if (entries.length > 0) {
      entries.reduce((prev, curr) => {
        if (this._similarItems(prev, curr)) {
          let score = curr.score * penalty;
          penalty -= 0.2;
          curr.score = score;
        }
        penalty = 0.8;
        return curr;
      });
    }
    return entries;
  }

```

So essentially you can use reduce here, since you're comparing to previous values in the array and modifying it. I tested it exactly like this and it still passes the ```Baseline.test.js``` tests